apps/prairielearn/src/lib/ltiOutcomes.ts

```diff
@@ -54,28 +57,42 @@ function xmlReplaceResult(sourcedId: string, score: number, identifier: string)
 export async function updateScore(assessment_instance_id: string) {
   if (assessment_instance_id == null) return;
 
-  const scoreResult = await sqldb.queryZeroOrOneRowAsync(sql.get_score, {
-    ai_id: assessment_instance_id,
-  });
-  if (scoreResult.rowCount === 0) return null;
-
-  const info = scoreResult.rows[0];
+  const info = await sqldb.queryOptionalRow(
+    sql.get_score,
+    {
+      ai_id: assessment_instance_id,
+    },
+    z.object({
+      ...AssessmentInstanceSchema.shape,
+      ...LtiOutcomeSchema.shape,
+      ...LtiCredentialSchema.shape,
+    }),
+  );
+  if (info === null) return null;
 
-  let score = info.score_perc / 100;
+  let score = (info.score_perc ?? 0) / 100;
```

Comment:
As pointed out by @jonatanschroeder , `null / 100 = 0`, so I preserve that behavior.